qualityassurance/webqualityassurancehandbook.md

```diff
@@ -0,0 +1,162 @@
+
+
+It should be used as a guide to ensure a great & consistent experience for the end user.
+
+
+### Blackbox
+
+#### General
+-  All UI scales well and looks good on small form factor devices
+-  All errors are handled and UI reacts appropriately
+-  Errors are clear in meaning & reflect what happened
+-  Activity indicators are used when loading data async
+-  Empty information is handled correctly in UI (optional fields etc)
+-  Long running operations give a indication of progress if possible
+-  Breadcrumbs are used where appropriate
+
+
+#### Navbar
+-  UI reflects the current active page
+-  Icon is appropriate for the page
+
+
+#### Forms
+-  Invalid inputs correctly trigger validation errors
+-  All form inputs have associated labels
+-  Appropriate validation for different inputs (phone number, postcode)
+-  The forms are not too long & and are grouped logically
+-  The user is alerted of the form lifecycle (success, error)
+-  When modals are used in the context of updates, ensure inputs are correctly pre populated.
+-  Autofill is disabled on inputs where appropriate
+-  Inputs have correct type property ( emails have email type, phone numbers have tel type)
+
+
+#### Modals
+-  Modals are animated on both display and dismiss
+-  Modals are used where appropriate over pages
+
+
+#### Formatting
+-  Dates are formatted correctly for the appropriate audience.
+-  Monetary values are formatted correctly for the appropriate audience
+-  Phone numbers are formatted correctly for the appropriate audience
+
+
+#### Pagination
+-  UI correctly indicates current page
+-  UI shows at least 2 page tabs ahead and below the current page
+
+
+#### Login & Auth
+-  User can only access pages that require 
+-  Authentication when they are logged in
+-  Login & Logout works as expected
+-  Reset password works as expected
+
+
+#### Registration
+-  Email addresses are verified & unique (if applicable)
+-  Passwords meet minimum standards (min 8 chars)
+-  Passwords should not have maximum length restrictions (allow at least 255 characters).
+
+
+#### Emails
+-  Emails look good on small form factor devices
+-  Ensure correct subject & sender name
+-  Correct email is sent, information is conveyed appropriately
+-  Emails are received in a timely manner
+-  Emails come from the appropriate email (often the clients)
+-  Links point to the correct address
+
+
+#### Push Notifications
+-  Push notifications are received in a timely manner
+-  They show correct & relevant information
+
+
+#### Permissions & Roles
+-  Ensure permissions & roles match up to possible actions (only admins can create users etc)
+
+
+### Whitebox
+
+#### General
+-  Filenames must follow standard coding conventions (UsersController.php)
```

Comment:
There are two whitespaces between the `-` and the statement on each bulletpoint - this should probably be moved to just one